scalameta/semantic/shared/src/main/scala/scala/meta/internal/semantic/Conversion.scala

```diff
@@ -0,0 +1,143 @@
+package scala.meta
+package internal.semantic
```

Comment:
I have reverted these changes and kept the conversion in package.scala. I'd like to move them out of the package object eventually, nesting has gotten out of hand

-  package object
- Xtension
- def
- .map
- object Extractor
- .unapply
- match
- .map 
- match

The quickest win would be to refactor

```scala
// before
def convert(s.Db): m.Db = {
  val mdb = sdb.entries.map {
     // ... hundreds of liens
  }
  m.Db(mdb)
}

// after
def convert(s.Db): m.Db = {
  val mdb = sdb.entries.map(convertEntry)
}
def convertEntry(s.Attributes): m.Attributes = {
  // ....
}
```